### PR TITLE
SL-18837 set build id based on github workflow run id

### PR DESCRIPTION
**Pull request by brad-linden**

### CODE REVIEW

**Review by nat-goodspeed — APPROVED** [review]
I like it. But I guess LLVersionInfo needs to be updated also.

Empirically, `$GIT_REF` is (or at least contains) the branch, e.g. "Second Life Test actions". If for a tag build it expands to the tag name, that would let us use a tag naming convention to set VIEWER_CHANNEL according to our existing practice.

File: indra/newview/llappviewer.cpp
Comment (nat-goodspeed) [inline indra/newview/llappviewer.cpp:3555]: Please use lowercase `stringize()` instead of the `STRINGIZE()` macro for new usage.

`STRINGIZE()` predates (type-safe) support for variadic functions. It accepts a partial `operator<<()` expression, which works but is eyebrow-raising for future maintainers. Besides, it upsets Dr. Stroustrup.

`stringize()` is a proper variadic function.

For a single argument like this, you could alternatively use `std::to_string()`.

File: indra/llcommon/llprocessor.cpp
Comment (nat-goodspeed) [inline indra/llcommon/llprocessor.cpp:749]: I admit I'm curious about the circumstances under which this fails, and why it's in this particular PR?

**Comment by nat-goodspeed** [comment]
This Windows build failed in a way reminiscent of the problem that caused me all the flailing with `py.exe`: `Access is denied` errors trying to run `C:/hostedtoolcache/windows/Python/3.11.4/x64/python.exe`. But you cut to the chase somehow, undoing the `py.exe` changes and getting clean builds anyway. What did you do then, and why isn't it working now?

File: .github/workflows/build.yaml
Comment (nat-goodspeed) [inline .github/workflows/build.yaml:96]: Hmm, I'm no longer sure `github.run_id` is the right variable. Consider https://github.com/secondlife/viewer/actions/runs/5205525429. What I think we want is version 6.6.13.5205525429 so we could just paste that last component into the .../viewer/actions/runs/ URL. Instead the viewer versions are 231482258 (64-bit viewers) or 231582302 (32-bit viewer).

File: indra/newview/llappviewer.cpp
Comment (brad-linden) [reply indra/newview/llappviewer.cpp:3555]: good point, thanks

File: indra/llcommon/llprocessor.cpp
Comment (brad-linden) [reply indra/llcommon/llprocessor.cpp:749]: I've been running with this workaround locally for a while now when building on Apple Silicon and running unit tests under rosetta2.  I hoped to get it into the actions branch in advance of github making Apple Silicon M1 runner hosts available shortly so we can use them.  It is unrelated to the other changes here, but will be needed once we have M1 based build runners.

However, I don't know what the right fix is here.  I can't yet find the docs for apple's machdep.cpu.feature_bits contents, but it is unsurprising to me that this will have a new unexpected value when running under rosetta2.

**Comment by brad-linden** [comment]
> This Windows build failed in a way reminiscent of the problem that caused me all the flailing with `py.exe`: `Access is denied` errors trying to run `C:/hostedtoolcache/windows/Python/3.11.4/x64/python.exe`. But you cut to the chase somehow, undoing the `py.exe` changes and getting clean builds anyway. What did you do then, and why isn't it working now?

yes.  previously the fix came from using the setup_python action as documented by github's docs.  I'm concerned that that fix is no longer working...

File: .github/workflows/build.yaml
Comment (brad-linden) [reply .github/workflows/build.yaml:96]: Yeah you're right that's not what I expected from reading the docs.  I'll keep investigating this

Comment (brad-linden) [reply .github/workflows/build.yaml:96]: correction.  github.run_id is properly matching with that URL component as I read it.  for example: https://github.com/secondlife/viewer/actions/runs/5384282553 

even though those builds are currently failing due to the other issues described in this PR, they do properly set the build to be 6.6.13.5384282553 for all 3 builds.

**Comment by brad-linden** [comment]
I have removed all the prior commits trying to work around python build failures.  They are resolved after the latest merge with the main [actions](https://github.com/secondlife/viewer/tree/actions) branch

File: indra/llcommon/llrand.cpp
Comment (nat-goodspeed) [inline indra/llcommon/llrand.cpp:96]: Couldn't the whole body of `ll_internal_random_float()` be:
```
inline F32 ll_internal_random_float()
{
    return F32(ll_internal_random_double());
}
```

File: .github/workflows/build.yaml
Comment (nat-goodspeed) [reply .github/workflows/build.yaml:92]: I do see that in your most recent branch build, https://github.com/secondlife/viewer/actions/runs/5559609923 in fact produces viewer version 6.6.13.5559609923, which is what I was hoping for. Not sure what went wrong with the build I first cited above.

**Review by nat-goodspeed — APPROVED** [review]

File: .github/workflows/build.yaml
Comment (brad-linden) [reply .github/workflows/build.yaml:20]: This was a bit superstitious of me.  I had seen builds before where a matrix variable was not properly being inherited as I expected it in one of these include branches.  I'll read up on the docs to be sure, and verify that this is unneeded and remove it

File: .github/workflows/build.yaml
Comment (brad-linden) [reply .github/workflows/build.yaml:92]: it looks like just a minor confusion.  double checking now, the change 3ffa57473d4aec6a32807873513f7d880ec4bc67 that it was built from preceded this branch by a couple commits

File: indra/llcommon/llrand.cpp
Comment (brad-linden) [reply indra/llcommon/llrand.cpp:96]: oh boy do I have some simplifications in mind for all this.  I should probably write a jira

**Pull request merged**
